bool isStop
float32 distance
